### ISSUE TITLE
Global search

### ISSUE THREAD

**Issue by ramingar** [Gisto/Gisto#168]
It would be amazing if Gisto could look for your gists in any place in your desktop (similar what Dash for OSX do).

**Comment by morsdyce** [comment]
I'm not sure I understand what you mean. I looked at Dash for OSX website and it didn't mention anything other than a regular search.

Could you provide a screenshot or a screencast of what feature are you talking about?

Thanks

**Comment by ramingar** [comment]
Dash runs in background and keeps listen for everything/anywhere you write. If you write "singleton\`" (an abbreviation pacted before) and Dash finds a snippet with that abbreviation, it proposes the snippet found in a modal window.

![abbreviation](http://www.lafermeduweb.net/images/billets/2013-11/snippetInDash.png)

![snippet showed](https://kapeli.com/img/snippetInUse.png)

An example video: https://youtu.be/doeDujYZkQY?t=1m15s (it happens when he writes "sprite\`" in TextEdit).

* * *

I don't know if this is possible... Do you think it would be a more realistic approach that Gisto showed a dialog to look for your snippets when you type a combo key (e.g. ctrl+space)? 
(Then, maybe "global shortcuts" would be a better name for this issue).

* * *

Thank you very much for your time!

**Comment by morsdyce** [comment]
This is actually something I wanted to do in the past but it increases in complexity allot as we are a cross platform application.

This is allot easier to do on OSX alone but supporting windows and linux would be allot harder and probably require us to do c++ modules that we're currently not well versed in.

I hope sometime in the future we'll be able to get to it and make it work but it won't be anytime soon.

Thanks

**Comment by ramingar** [comment]
That would be a most appreciated feature. Make sure after you implement this feature I would be very glad if I invite you a beer XDD.

**Comment by sanusart** [comment]
@ramingar @morsdyce 
I am using [Zeal](http://zealdocs.org/) on Linux sometimes (has Windows support too).
As far as I see/understand - both need generated "docsets". 
We need to look into generating "docset" for both, Dash and Zeal. And while Zeal can't listen to system events, use of plug-ins might be needed.

**Comment by ramingar** [comment]
@sanusart @morsdyce 
Dash supports adding your own snippets (also generated docsets).
I use Zeal too. Zeal listen for global shortcuts to be showed.

Btw, Gisto shine magnificently editing and showing your code snippets, thus I don't think it was necessary to browse offline documentation. But it's not too pleasant to have to switch into Gisto, search your snippet, copy, switch to the application where you were working and paste it. It's not an arduous task at all, but it would be a lot more enjoyable task if you can show and use your snippet wherever you were writing... simply after you write an abbreviation (like Dash) or type a global shortcut (like Zeal).

What's your point?

**Comment by morsdyce** [comment]
I think he meant adding support from gisto to dash/zeal which i am not in favor of.

Zeal is open source so we can look into how it uses global shortcuts and see if we can make a solution to work with node-webkit that will work across all platforms.

I have also thought of writing IDE plugins for popular ones such as Intellij platform/sublime but it would take allot of work specially as we're not fluent in java or python.

We need to review this again later on once we finish our next big feature (offline access) and see if this is feasible.

**Comment by sanusart** [comment]
Yes that's what I meant, but I miss read the whole issue (I do that
sometimes). So ignore my last comment. And I don't see value of Gisto
integrated into dash/zeal like doc browser.

Back to the issue, seems nwjs global shortcuts listening are possible, at
least according to https://github.com/nwjs/nw.js/wiki/Shortcut

**Comment by morsdyce** [comment]
Oh nice, I haven't seen this api.
I don't think it can paste into the app you were focused in but we could copy the content to clipboard to make it easier to paste in.

Maybe we can create a new window with an interface like alfred to search for a specific gist and copy the entire content/specific file to the clipboard.

**Comment by ramingar** [comment]
That would be REALLY AWESOME!!
Thanks you both.

**Comment by sanusart** [comment]
Or maybe shortcut API can then trigger tray icon API to pop interactive  notifications.
But yes, lets concentrate on #5 first.

**Comment by MVECoder** [comment]
Jumping into this zombie ticket because a primary goal for Gisto is to help manage snippets of code.  Being able to search through the code is a significant feature that is missing from the current version(s) of Gisto.  The current search functionality is only allowing a basic snippet title and tag search and is restrictive.  I go to github's site in order to find actual code blocks by search, defeating the purpose of this fantastic tool.

**Comment by sanusart** [comment]
@MVECoder This issue is a bit different from what you describe. This issue is not dealing with searching within gist files (which is a different problem).
